dlls/winevulkan/make_vulkan

```diff
@@ -2060,52 +2307,92 @@ class FreeFunction(object):
-                elif m.needs_conversion():
-                    LOGGER.error("Unhandled conversion for {0}".format(m.name))
+                        unwrap = m.free("in[i].", False)
+                        if convert == unwrap:
+                            body += "        " + unwrap
+                        elif unwrap == "":
+                            body += "#if defined(USE_STRUCT_CONVERSION)\n        {0}#endif /* USE_STRUCT_CONVERSION */\n".format(convert)
+                        else:
+                            body += "#if defined(USE_STRUCT_CONVERSION)\n        {0}#else\n        {1}#endif /* USE_STRUCT_CONVERSION */\n".format(convert, unwrap)
             body += "    }\n"
         else:
             body += "    if (!in) return;\n\n"
 
         body += "    free(in);\n"
 
         body += "}\n\n"
+
+        if self.operand.needs_unwrapping():
+            body += "#if defined(USE_STRUCT_CONVERSION)\n"
+
         return body
 
     def _generate_free_func(self):
         # E.g. VkCommandBufferBeginInfo.pInheritanceInfo needs freeing.
-        if not self.struct.needs_free():
+        if not self.operand.needs_free():
             return ""
 
+        if not isinstance(self.operand, VkStruct):
+            return ""
+
+        body = ""
+
         # Generate function prototype.
-        body = "static inline void {0}({1}_host *in)\n{{\n".format(self.name, self.type)
+        body += "static inline void {0}({1}_host *in)\n{{\n".format(self.name, self.type)
 
-        for m in self.struct:
-            if m.needs_conversion() and m.is_dynamic_array():
-                count = m.dyn_array_len if isinstance(m.dyn_array_len, int) else "in->{0}".format(m.dyn_array_len)
-                if m.is_const():
-                    # Add a cast to ignore const on conversion structs we allocated ourselves.
-                    body += "    free_{0}_array(({0}_host *)in->{1}, {2});\n".format(m.type, m.name, count)
+        for m in self.operand:
+            if m.needs_free():
+                convert = m.free("in->", True)
+                if self.operand.needs_conversion() and not self.operand.needs_unwrapping():
+                    body += "    " + convert
                 else:
-                    body += "    free_{0}_array(in->{1}, {2});\n".format(m.type, m.name, count)
+                    unwrap = m.free("in->", False)
+                    if convert == unwrap:
+                        body += "    " + unwrap
+                    elif unwrap == "":
+                        body += "#if defined(USE_STRUCT_CONVERSION)\n    {0}#endif /* USE_STRUCT_CONVERSION */\n".format(convert)
+                    else:
+                        body += "#if defined(USE_STRUCT_CONVERSION)\n    {0}#else\n    {1}#endif /* USE_STRUCT_CONVERSION */\n".format(convert, unwrap)
 
         body += "}\n\n"
+
+        if self.operand.needs_unwrapping():
+            body += "#if defined(USE_STRUCT_CONVERSION)\n"
```

Comment:
Not a fan of how the PP #ifs are handles, every free/conversion should handle it on their own.
e.g. if it does both conversion and unwrapping
```c
#if defined(USE_STRUCT_CONVERSION)
return_value_host convert_to_host(params)
#else
return_value convert_to_host(params)
#endif 
{
   //body
}
```  
or if it does only alignment conversion

```c
#if defined(USE_STRUCT_CONVERSION)
return_value_host convert_to_host(params)
{
   //body
}
#endif
```

or if it only does unwrapping 

```c
return_value convert_to_host(params)
{
   //body
}
```